clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10778,6 +10778,101 @@ def warn_imp_cast_drops_unaligned : Warning<
   "implicit cast from type %0 to type %1 drops __unaligned qualifier">,
   InGroup<DiagGroup<"unaligned-qualifier-implicit-cast">>;
 
+def warn_func_effect_allocates : Warning<
+  "'%0' function '%1' must not allocate or deallocate memory">,
+  InGroup<FunctionEffects>;
+
+def note_func_effect_allocates : Note<
+  "'%1' cannot be inferred '%0' because it allocates/deallocates memory">;
```

Comment:
Yeah, the code generating warnings/notes always puts the effect before the function name, so there's consistency there and inconsistency here:
```
def warn_func_effect_allocates : Warning<
  "'%0' function '%1' must not allocate or deallocate memory">,
  InGroup<FunctionEffects>;

def note_func_effect_allocates : Note<
  "'%1' cannot be inferred '%0' because it allocates/deallocates memory">;
```

Reply:
Hmm, I’m not sure I follow; `%0` is the first thing you pass to the diagnostic, `%1` the second, so swapping `%0` and `%1` in the note and swapping the order in which you pass the args to the diagnostic should work, or am I missing something?

Reply:
I decided I preferred inconsistency in DiagnosticSemaKinds.td than in the code that emits the diags:

```
        S.Diag(Diag.Loc, diag::warn_func_effect_allocates)
            << effectName << TopFuncName;
...
            S.Diag(Diag2.Loc, diag::note_func_effect_allocates)
                << effectName << CalleeName;
```

But maybe that's just a bit precious. As I was trying to word the diagnostics, there were times when they didn't always come out in consistent orders and I had bugs in passing the parameters in the right order.

Reply:
Another way to put it: when one reads through the ~100 lines of diagnostics in DiagnosticSemaKinds.td, I find it helpful that %0 is always the name of the effect and %1 is always a function (or block) name.

Reply:
These diagnostics aren't part of this branch any more.